webv2/web.go

```diff
@@ -1026,26 +1142,131 @@ func canRenameOrChangeType(colName, table string) (int, error) {
 }
 
 func checkPrimaryKeyPrefix(table string, refTable string, fk ddl.Foreignkey, tableInterleaveStatus *TableInterleaveStatus) bool {
-	sessionState := session.GetSessionState()
 
+	sessionState := session.GetSessionState()
 	childPks := sessionState.Conv.SpSchema[table].Pks
 	parentPks := sessionState.Conv.SpSchema[refTable].Pks
-	if len(childPks) >= len(parentPks) {
-		for i, pk := range parentPks {
-			if i >= len(fk.ReferColumns) || pk.Col != fk.ReferColumns[i] || pk.Col != childPks[i].Col || fk.Columns[i] != fk.ReferColumns[i] {
-				return false
+
+	interleaved := []ddl.IndexKey{}
+
+	for i := 0; i < len(parentPks); i++ {
+
+		for j := 0; j < len(childPks); j++ {
+
+			for k := 0; k < len(fk.ReferColumns); k++ {
+
+				if parentPks[i].Col == childPks[j].Col && parentPks[i].Col == fk.ReferColumns[k] && childPks[j].Col == fk.ReferColumns[k] {
+
+					interleaved = append(interleaved, parentPks[i])
+				}
 			}
+
 		}
-	} else {
-		return false
+
 	}
-	return true
+
+	diff := []ddl.IndexKey{}
+
+	if len(interleaved) == 0 {
+
+		for i := 0; i < len(parentPks); i++ {
+
+			for j := 0; j < len(childPks); j++ {
+
+				if parentPks[i].Col != childPks[j].Col {
+
+					diff = append(diff, parentPks[i])
+				}
+
+			}
+		}
+
+	}
+
+	caninterleaved := []string{}
+	for i := 0; i < len(diff); i++ {
+
+		str := contains(fk.ReferColumns, diff[i].Col)
+
+		caninterleaved = append(caninterleaved, str)
+	}
+
+	if len(caninterleaved) > 0 {
+
+		for i := 0; i < len(caninterleaved); i++ {
+
+			sessionState := session.GetSessionState()
+
+			schemaissue := []internal.SchemaIssue{}
+
+			schemaissue = sessionState.Conv.Issues[table][caninterleaved[i]]
+
+			schemaissue = Remove(schemaissue, internal.Interleaved_Order)
+			schemaissue = Remove(schemaissue, internal.Interleaved_NotINOrder)
+			schemaissue = Remove(schemaissue, internal.Interleaved_ADDCOLUMN)
+
+			schemaissue = append(schemaissue, internal.Interleaved_ADDCOLUMN)
+
+			if len(schemaissue) > 0 {
+
+				if sessionState.Conv.Issues[table][caninterleaved[i]] == nil {
+
+					s := map[string][]internal.SchemaIssue{
+						caninterleaved[i]: schemaissue,
+					}
+					sessionState.Conv.Issues = map[string]map[string][]internal.SchemaIssue{}
+
+					sessionState.Conv.Issues[table] = s
+
+				} else {
+					sessionState.Conv.Issues[table][caninterleaved[i]] = schemaissue
+				}
+
+			}
+
+		}
+	}
+
+	if len(interleaved) > 0 {
+		return true
+	}
+
+	return false
+}
+
+func contains(fc []string, col string) string {
```

Comment:
Same as Remove() - This method is same as in primarykey.go , move the method to a common utility folder under webv2 and call it from both the places